context/app/static/js/components/detailPage/ProcessedData/ProcessedDataset/hooks.ts

```diff
@@ -68,18 +67,28 @@ export function useProcessedDatasetDetails(uuid: string) {
   return { datasetDetails, isLoading };
 }
 
-function processDatasetLabel(dataset: ProcessedDatasetInfo, hits: Required<SearchHit<ProcessedDatasetInfo>>[]) {
-  const multipleHitsWithSamePipeline = hits.filter((h) => h._source.pipeline === dataset.pipeline).length > 1;
-  const multipleHitsWithSamePipelineAndStatus =
-    multipleHitsWithSamePipeline &&
-    hits.filter((h) => h._source.pipeline === dataset.pipeline && h._source.status === dataset.status).length > 1;
-  if (multipleHitsWithSamePipelineAndStatus) {
-    return `${dataset.pipeline} (${dataset.status}) [${dataset.hubmap_id}]`;
+export function processDatasetLabel(
+  dataset: Pick<ProcessedDatasetInfo, 'assay_display_name' | 'pipeline' | 'status' | 'hubmap_id'>,
+  hits: { _source: Pick<ProcessedDatasetInfo, 'assay_display_name' | 'pipeline' | 'status'> }[],
+) {
+  const label = dataset.pipeline ?? dataset.assay_display_name[0];
+  const multipleHitsWithSameLabel =
+    hits.filter((h) => (h._source.pipeline ?? h._source.assay_display_name[0]) === label).length > 1;
+
+  const multipleHitsWithSameLabelAndStatus =
+    multipleHitsWithSameLabel &&
```

Comment:
We need to check for length. Empty arrays are truthy. In a short-circuit you should also wrap the array in `Boolean` or make a comparison to avoid returning `0`.

Reply:
Ahh I see - in this case, `multipleHitsWithSameLabel` and `multipleHitsWithSameLabelAndStatus` are booleans, not arrays. The check for length is done at assignment. Is it maybe preferable to store the arrays themselves to make the second check more efficient?

Reply:
Ah, my apologies I missed the checks against `.length`. Can we change the boolean variable names to include `has` for comprehension?

Reply:
Yes, good idea!